cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -396,11 +397,16 @@ void RegisterScalarAggregateBasic(FunctionRegistry* registry) {
   aggregate::AddBasicAggKernels(aggregate::SumInit, FloatingPointTypes(), float64(),
                                 func.get());
   // Add the SIMD variants for sum
+  auto cpu_info = arrow::internal::CpuInfo::GetInstance();
 #if defined(ARROW_HAVE_RUNTIME_AVX2)
```

Comment:
Is it possible to drop these #ifdef ?

Reply:
No. Other arch build like ARM/PPC will not have AddSumAvx2AggKernels function.